lib/rubocop/cop/style/comment_annotation.rb

```diff
@@ -13,25 +13,34 @@ class CommentAnnotation < Cop
 
         def investigate(processed_source)
           processed_source.comments.each do |comment|
-            margin, first_word, colon, space, note = split_comment(comment)
+            _margin, first_word, colon, space, note = split_comment(comment)
             next unless annotation?(comment) &&
                         !correct_annotation?(first_word, colon, space, note)
 
-            start = comment.loc.expression.begin_pos + margin.length
-            length = first_word.length + colon.to_s.length + space.to_s.length
-            range = Parser::Source::Range.new(processed_source.buffer,
-                                              start,
-                                              start + length)
-            add_offense(range, range)
+            add_offense(comment, :expression)
           end
         end
 
         private
 
-        def autocorrect(range)
+        def autocorrect(comment)
+          margin, first_word, colon, space, note = split_comment(comment)
+          start = comment.loc.expression.begin_pos + margin.length
+
           lambda do |corrector|
-            annotation_keyword = range.source.split(/:?\s+/).first
-            corrector.replace(range, annotation_keyword.upcase << ': ')
+            if note.nil?
+              start += first_word.length
+              range = Parser::Source::Range.new(comment,
+                                                start,
+                                                comment.loc.expression.end_pos)
+              corrector.remove(range)
+            else
+              length = first_word.length + colon.to_s.length + space.to_s.length
+              range = Parser::Source::Range.new(comment.loc.expression.source,
```

Comment:
I tested it out, and I see what you mean about the first parameter not really mattering. I am used to supplying some sort of source for the `Range` and that seems to usually be the `source`. I can go ahead and change this.